src/Message/AbstractRequest.php

```diff
@@ -136,4 +150,62 @@ public function setSecret($secret)
 
         return $this;
     }
+
+    /**
+     * @param string $baseUrl
+     *
+     * @return $this
+     */
+    public function setBaseUrl($baseUrl)
+    {
+        $this->setParameter('base_url', $baseUrl);
+
+        return $this;
+    }
+
+    /**
+     * @param string $method
+     * @param string $url
+     * @param mixed  $data
+     *
+     * @return Response
+     */
+    protected function sendRequest($method, $url, $data)
+    {
+        if (RequestInterface::GET === $method) {
+            return $this->httpClient->createRequest(
+                $method,
+                $this->getBaseUrl().$url,
+                null,
+                null,
+                $this->getRequestOptions()
+            )->send();
+        }
+
+        return $this->httpClient->createRequest(
+            $method,
+            $this->getBaseUrl().$url,
+            ['Content-Type' => 'application/json'],
+            json_encode($data),
+            $this->getRequestOptions()
+        )->send();
+    }
+
+    /**
+     * @return array
+     */
+    protected function getRequestOptions()
+    {
+        return ['auth' => [$this->getMerchantId(), $this->getSecret()]];
+    }
+
+    /**
+     * @param Response $response
+     *
+     * @return array
+     */
+    protected function getResponseBody(Response $response)
+    {
+        return empty($response->getBody(true)) ? [] : $response->json();
```

Comment:
uncatched `RuntimeException` will be thrown whenever `json_decode` fails (e.g. malformed json string).

Trivial: The `getBody` check seems to be unnecessary as `$response->json();` does same check.